### DISCUSSION TITLE
使用了 RecursionField做递归渲染，但字段联动的内置表达式，如 $self 不生效

### DISCUSSION THREAD

**Discussion by li9269391** [alibaba/formily#1596] [Q&A 2.x] [answered]
例子：https://codesandbox.io/s/heuristic-darkness-7zu6i
![image](https://user-images.githubusercontent.com/10287877/122372519-4fbf2080-cf93-11eb-985f-b8b6fcc2a5ff.png)

**Comment by janryWang** [comment, marked as answer]
为什么要这么用呢？是要解决什么需求

**Reply by li9269391** [reply]
最后再次确认是我想复杂了，用错场景

**Comment by xijin** [comment]
我也想问下这种情况下怎么处理，我是想做一个纯schema驱动的组件容器，x-reactions联动关系配置也是组件容器的一部分

**Reply by li9269391** [reply]
你直接用就好了，
```
<FormProvider form={normalForm}>
      <SchemaField schema={schema} />
</FormProvider>
```
我之前是想做二次封装，做内部的结构转换等内置需求。其实不是这样用，后面只需要处理 @formily/core 的模型实例和使用 @formily/react 的 hooks 就能满足
